src/overlays/actors/ovl_En_Fish/z_en_fish.c

```diff
@@ -47,73 +63,709 @@ static ColliderJntSphInit D_80A17040 = {
+        osSyncPrintf("Warning : dousa 3 消滅 が呼ばれずにデモが終了した(%s %d)(arg_data 0x%04x)\n", "../z_en_sakana.c",
+                     1169, this->actor.params);
+        EnFish_ClearCutsceneData(this);
+        Actor_Kill(&this->actor);
+        return;
+    }
+
+    this->slowPhase += 0x111;
+    this->fastPhase += 0x500;
+
+    switch (csAction->action) {
+        case 1:
+            EnFish_Cutscene_FlopOnGround(this, globalCtx);
+            break;
+        case 2:
+            EnFish_Cutscene_WiggleFlyingThroughAir(this, globalCtx);
+            break;
+        case 3:
+            // DEMO fish termination
+            osSyncPrintf("デモ魚消滅\n");
+            EnFish_ClearCutsceneData(this);
+            Actor_Kill(&this->actor);
+            return;
+        default:
+            // Improper DEMO action
+            osSyncPrintf("不正なデモ動作(%s %d)(arg_data 0x%04x)\n", "../z_en_sakana.c", 1200, this->actor.params);
+            break;
+    }
+
+    startPos.x = csAction->startPos.x;
+    startPos.y = csAction->startPos.y;
+    startPos.z = csAction->startPos.z;
+    endPos.x = csAction->endPos.x;
+    endPos.y = csAction->endPos.y;
+    endPos.z = csAction->endPos.z;
+
+    progress = func_8006F93C(csAction->endFrame, csAction->startFrame, globalCtx->csCtx.frames);
+
+    this->actor.world.pos.x = ((endPos.x - startPos.x) * progress) + startPos.x;
+    this->actor.world.pos.y = ((endPos.y - startPos.y) * progress) + startPos.y + D_80A17014;
+    this->actor.world.pos.z = ((endPos.z - startPos.z) * progress) + startPos.z;
+
+    this->actor.floorHeight = BgCheck_EntityRaycastFloor4(&globalCtx->colCtx, &this->actor.floorPoly, &bgId,
+                                                          &this->actor, &this->actor.world.pos);
+}
+
+// Update functions and Draw
+
+void EnFish_OrdinaryUpdate(EnFish* this, GlobalContext* globalCtx) {
+    if (this->timer > 0) {
+        this->timer--;
+    }
+
+    this->slowPhase += 0x111;
+    this->fastPhase += 0x500;
+
+    if ((this->actor.child != NULL) && (this->actor.child->update == NULL) && (&this->actor != this->actor.child)) {
+        this->actor.child = NULL;
+    }
+
+    if ((this->actionFunc == NULL) || (this->actionFunc(this, globalCtx), (this->actor.update != NULL))) {
+        Actor_MoveForward(&this->actor);
+
+        if (this->unk_250 != 0) {
+            Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 17.5f, 4.0f, 0.0f, this->unk_250);
+        }
+
+        if (this->actor.xzDistToPlayer < 70.0f) {
+            CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+        }
+
+        Actor_SetFocus(&this->actor, this->actor.shape.yOffset * 0.01f);
+
+        if (Actor_HasParent(&this->actor, globalCtx)) {
+            this->actor.parent = NULL;
+
+            if (this->actor.params == FISH_DROPPED) {
+                Actor_Kill(&this->actor);
+                return;
+            }
+
+            EnFish_BeginRespawn(this);
```

Comment:
```suggestion
            if (this->actor.params == FISH_DROPPED) {
                Actor_Kill(&this->actor);
            }
            else {
                EnFish_BeginRespawn(this);
            }
```

Reply:
We usually do early returns after `Actor_Kill`.